pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -820,6 +843,39 @@ public String generateUniqueProducerName() {
     public AuthenticationService getAuthenticationService() {
         return authenticationService;
     }
+    
+    private int getLookupRequestPermits() {
+        int pendingLookupRequest = pulsar.getConfiguration().getMaxConcurrentLookupRequest();
+        try {
+            Optional<Map<String, String>> data = dynamicConfigurationCache.get(LOOKUP_THROTTLING_PATH);
```

Comment:
yes, we can read from config-file but part of this PR also requires to test updating semaphore at runtime and if config value gets changed then we are also updating ```lookupRequestSemaphore``` so, even with Dynamic-config-change we might need this listener logic and we have to provide a way to inject listener on config value change in dynamic-config-PR.
So, should we keep this PR open until we come up with the approach as anyway it should be part of 1.17 release.